### ISSUE TITLE
Failing InternalsVisibleToAndStrongNameTests

### ISSUE THREAD

**Issue by AdamSpeight2008** [dotnet/roslyn#17929]
Version Used:  #17906

`InternalsVisibleToAndStrongNameTests`

- `IVTDeferredFailKeyMismatch`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```

- `IVTDeferredFailSignMismatch`
```
Message: 
MISSING ERROR MESSAGES:
BC36958: Friend access was granted by 'Paul, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null', but the strong name signing state of the output assembly does not match that of the granting assembly.
UNEXPECTED ERROR MESSAGES:
BC36957: Friend access was granted by 'Paul, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null', but the public key of the output assembly does not match that specified by the attribute in the granting assembly.
BC36981: Error extracting public key from container 'roslynTestContainer': Keyset does not exist (Exception from HRESULT: 0x80090016)

Expected: True
Actual:   False
```

- `IVTDeferredSuccess`
```
Message: Assert.True() Failure
Expected: True
Actual:   False
```

- `IVTSigned`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_InaccessibleSymbol2, "C").WithArguments("C", "Friend").WithLocation(2, 26),
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_InaccessibleSymbol2, "C").WithArguments("C", "Friend").WithLocation(2, 26),
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```

- `IVTSignedThroughIAssembly`
```
Message: Assert.True() Failure
Expected: True
Actual:   False
```

- `KeyContainerFromAttributes_PublicSign`
```
Message: 
MISSING ERROR MESSAGES:
BC42379: Attribute 'System.Reflection.AssemblyKeyNameAttribute' is ignored when public signing is specified.
UNEXPECTED ERROR MESSAGES:
BC42379: 

Expected: True
Actual:   False
```

- `KeyFileFromAttributes_PublicSign`
```
Message: 
MISSING ERROR MESSAGES:
BC42379: Attribute 'System.Reflection.AssemblyKeyFileAttribute' is ignored when public signing is specified.
UNEXPECTED ERROR MESSAGES:
BC42379: 

Expected: True
Actual:   False
```

- `PublicKeyFromKeyContainerAttributes`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```

- `PubKeyFromKeyContainerAttribute`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```

- `PublicSign_PublicKeyFromOptions_PublicSign`
```
Message: 
MISSING ERROR MESSAGES:
BC42379: Attribute 'System.Reflection.AssemblyKeyFileAttribute' is ignored when public signing is specified.
BC42379: Attribute 'System.Reflection.AssemblyKeyNameAttribute' is ignored when public signing is specified.
UNEXPECTED ERROR MESSAGES:
BC42379: 
BC42379: 

Expected: True
Actual:   False
```

- `PublicSign_IgnoreSourceAttributes`
```
Message: 
MISSING ERROR MESSAGES:
BC42379: Attribute 'System.Reflection.AssemblyKeyFileAttribute' is ignored when public signing is specified.
BC42379: Attribute 'System.Reflection.AssemblyKeyNameAttribute' is ignored when public signing is specified.
UNEXPECTED ERROR MESSAGES:
BC42379: 
BC42379: 

Expected: True
Actual:   False
```

- `SignModuleKeyContainerAttr`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```

- `SignModuleKeyContainerCmdLine`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```
- `SignModuleKeyContainerCmdLine_1`
```
Message: 
Expected:
Actual:
    Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Diff:
++>     Diagnostic(ERRID.ERR_PublicKeyContainerFailure).WithArguments("roslynTestContainer", "Keyset does not exist (Exception from HRESULT: 0x80090016)").WithLocation(1, 1)
Expected: True
Actual:   False
```

**Comment by AdamSpeight2008** [comment]
We maybe able to fix these if we follow the C# implementation of these tests.

**Comment by AdamSpeight2008** [comment]
```vb
    <Fact>
    Public Sub IVTDeferredFailKeyMismatch()
        Dim s = <compilation name="Paul">
                    <file name="a.vb"><![CDATA[
'key is wrong in the first digit
<Assembly: System.Runtime.CompilerServices.InternalsVisibleTo("John, PublicKey=10240000048000009400000006020000002400005253413100040000010001002b986f6b5ea5717d35c72d38561f413e267029efa9b5f107b9331d83df657381325b3a67b75812f63a9436ceccb49494de8f574f8e639d4d26c0fcf8b0e9a1a196b80b6f6ed053628d10d027e032df2ed1d60835e5f47d32c9ef6da10d0366a319573362c821b5f8fa5abc5bb22241de6f666a85d82d6ba8c3090d01636bd2bb")>
Friend Class C
 Friend Sub Foo()
 End Sub
End Class
]]>
                    </file>
                </compilation>.Value

        Dim other = CreateCompilationWithMscorlib(s,
            options:=TestOptions.ReleaseDll.WithCryptoKeyFile(s_keyPairFile).WithStrongNameProvider(s_defaultProvider))

        other.VerifyDiagnostics()
' <--- OK so far

        Dim requestor = CreateCompilationWithMscorlibAndReferences(
<compilation name="John">
    <file name="a.vb"><![CDATA[
Imports MyC=C
<Assembly: System.Reflection.AssemblyKeyName("roslynTestContainer")>
Public Class A
End Class
]]>
    </file>
</compilation>,
        {New VisualBasicCompilationReference(other)},
        options:=TestOptions.ReleaseDll.WithStrongNameProvider(s_defaultProvider))

        Dim asm_PubKey = requestor.Assembly.Identity.PublicKey
' <-- Uninitialised PublicKey.

        Assert.True(ByteSequenceComparer.Equals(s_publicKey, asm_PubKey))
        requestor.VerifyDiagnostics(
            Diagnostic(ERRID.ERR_FriendRefNotEqualToThis).WithArguments("Paul, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ce65828c82a341f2")
            )

        'CompilationUtils.AssertTheseDiagnostics(requestor, <errors>BC36957: Friend access was granted by 'Paul, Version=0.0.0.0, Culture=neutral, PublicKeyToken=ce65828c82a341f2', but the public key of the output assembly does not match that specified by the attribute in the granting assembly.</errors>)

    End Sub
```

**Comment by gafter** [comment]
Does this still fail for you in master?

**Comment by AdamSpeight2008** [comment]
@gafter I'll check.

**Comment by AdamSpeight2008** [comment]
Only if I run the VS Cmd Prompt in Adminstrator mode, otherwise nope.

**Comment by AdamSpeight2008** [comment]
@gafter What aspect of those unit test would require administrator privileges?

**Comment by AdamSpeight2008** [comment]
@gafter Doesn't seem to help see recent merge of  #17906

**Comment by AdamSpeight2008** [comment]
@jaredpar I still get them, but only I don't use the `Run As Adminstrator` run option on the VS Cmd Prompt.
